### PR TITLE
Added support for Kotlin Data classes.

### PR DESCRIPTION
**Pull request by bodiam**
Should fix https://github.com/datafaker-net/datafaker/issues/918

### CODE REVIEW

File: src/test/kotlin/net/datafaker/example/SchemaExampleTest.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package net.datafaker.example
+
+import net.datafaker.Faker
+import net.datafaker.transformations.Field
+import net.datafaker.transformations.JavaObjectTransformer
+import net.datafaker.transformations.Schema
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import java.util.*
```
Comment (kingthorin) [inline src/test/kotlin/net/datafaker/example/SchemaExampleTest.kt:9]: Use specific imports?

**Review by kingthorin — COMMENTED** [review]
That's the only thing I notice but I'm also not up on maven build or kotlin code.

File: src/test/kotlin/net/datafaker/example/SchemaExampleTest.kt (hunk 1 above)
Comment (bodiam) [reply src/test/kotlin/net/datafaker/example/SchemaExampleTest.kt:9]: Why?

Comment (kingthorin) [reply src/test/kotlin/net/datafaker/example/SchemaExampleTest.kt:9]: I'm not overly picky about it. Might be a spotless thing, I'd have to double check. I just figured that'd be more consistent, but honestly I didn't check the existing code base to see.

Comment (bodiam) [reply src/test/kotlin/net/datafaker/example/SchemaExampleTest.kt:9]: Well, it's not me who brought it up :-) I have really no idea, I just use IntelliJ defaults, and that's it.

Comment (kingthorin) [reply src/test/kotlin/net/datafaker/example/SchemaExampleTest.kt:9]: That's fair, if that's how IntelliJ defaults then it likely is like that elsewhere in the code base. We can close this.

**Comment by bodiam** [comment]
I am not sure if this is a complete fix, it might still break if you have multiple constructors etc, but it does pass the test now.

**Pull request merged**